magpie/ui/management/views.py

```diff
@@ -815,11 +834,30 @@ def add_service(self):
         cur_svc_type = self.request.matchdict["cur_svc_type"]
         svc_types, cur_svc_type, services = self.get_services(cur_svc_type)
 
+        services_keys_sorted = self.get_service_types()
+        services_phoenix_indices = [(1 if services_keys_sorted[i] in register.SERVICES_PHOENIX_ALLOWED else 0)
+                                    for i in range(len(services_keys_sorted))]
+
+        return_data = {u"invalid_service_name": False, u"invalid_service_url": False, u"form_service_name": u"", u"form_service_url": u"", u"cur_svc_type": cur_svc_type, u"service_types": svc_types, u"services_phoenix": register.SERVICES_PHOENIX_ALLOWED, u"services_phoenix_indices": services_phoenix_indices}
```

Comment:
Don't redefine the whole dict, just update keys of one defined at func start.

Reply:
It was defined at the end of the function and moved up. 
It has been moved up to allow return_data to be sent without missing values when error occurs.